pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -110,6 +147,8 @@ func (c *RebalancerController) doWorkloadRebalance(ctx context.Context, rebalanc
 	newStatus = rebalancer.Status.DeepCopy()
 	if len(newStatus.ObservedWorkloads) == 0 {
 		newStatus = c.buildWorkloadRebalancerStatus(rebalancer)
+	} else {
+		newStatus = c.syncWorkloadsFromSpecToStatus(rebalancer)
```

Comment:
Maybe we don't need to check if it is first reconcile, just use `newStatus = c.syncWorkloadsFromSpecToStatus(rebalancer)` to gather observed items?

Reply:
you are right, just use `syncWorkloadsFromSpecToStatus` is enough.

however, in `syncWorkloadsFromSpecToStatus`, the resource list in `status` is disordered (previously is mainly following the input order). I prefer to sort them in dict order when just using `syncWorkloadsFromSpecToStatus`, as you can see in my latest push.

Reply:
sort +1.